core/geometry/src/curve/CurveChainWithDistanceIndex.ts

```diff
@@ -103,16 +103,18 @@ export class PathFragment {
   /**
    * Convert distance to local fraction and apply that to interpolate between the stored curve fractions.
    * Note that proportional calculation does NOT account for non-uniform parameterization in the child curve.
+   * Return 0 if conversion failed.
    */
   public chainDistanceToInterpolatedChildFraction(distance: number): number {
-    return Geometry.inverseInterpolate(
+    const value = Geometry.inverseInterpolate(
       this.childFraction0,
       this.chainDistance0,
       this.childFraction1,
       this.chainDistance1,
       distance,
       this.childFraction0,
-    )!; // the interval must have nonzero length so division should be safe
+    );
```

Comment:
Is there a reason you didn't just change the `!` to ` ?? 0`?

Reply:
will fix in the next commit